### PR TITLE
Cosmos Direct TCP port from azure-cosmosdb v2.6.1

### PR DESCRIPTION
**Pull request by David-Noble-at-work**
This port brings v3 into alignment with v2.6.1 of the Cosmos SDK for Java. Except for RntbdTransportClient.java which has some adjustments for Reactor the implementations are now equivalent.

### CODE REVIEW

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/Configs.java [hunk 1]
```diff
@@ -68,9 +76,11 @@ public SslContext getSslContext() {
     }
 
     public Protocol getProtocol() {
-        String protocol = getJVMConfigAsString(PROTOCOL, DEFAULT_PROTOCOL.toString());
+        String protocol = System.getProperty(PROTOCOL_PROPERTY, firstNonNull(
+            emptyToNull(System.getenv().get(PROTOCOL_ENVIRONMENT_VARIABLE)),
+            DEFAULT_PROTOCOL.name()));
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/Configs.java]: given the fact that protocol name is different in v2 and v3, Kushagra did some coding here to ensure our CI works for both v2 and v3. As v3 is on a different repo with new CIs, I am not sure how much of that is still applicable but good to check.

@kushagraThapar does this break anything in the CI?

Comment (David-Noble-at-work) [reply sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/Configs.java]: Thanks for the review. Kushagra may know something more but I've not seen that CI is broken by the change. l'll wait to hear back from him before resolving this issue.

File: sdk/cosmos/microsoft-azure-cosmos-benchmark/pom.xml [hunk 2]
```diff
@@ -43,28 +43,46 @@ Licensed under the MIT License.
...
+      <version>${metrics.version}</version>
     </dependency>
 
     <dependency>
       <groupId>io.dropwizard.metrics</groupId>
       <artifactId>metrics-graphite</artifactId>
+      <version>${metrics.version}</version>
     </dependency>
 
+    <dependency>
```
Comment (christopheranderson) [inline sdk/cosmos/microsoft-azure-cosmos-benchmark/pom.xml]: ✅ We have approval for these dependencies (in email)

File: sdk/cosmos/microsoft-azure-cosmos/pom.xml [hunk 3]
```diff
@@ -113,6 +113,18 @@ Licensed under the MIT License.
       <artifactId>guava</artifactId>
     </dependency>
 
+    <dependency>
```
Comment (christopheranderson) [inline sdk/cosmos/microsoft-azure-cosmos/pom.xml]: ✅ We have approval for these dependencies (in email)

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/BridgeInternal.java [hunk 4]
```diff
@@ -52,6 +53,10 @@ public static Document documentFromObject(Object document, ObjectMapper mapper)
         return Document.FromObject(document, mapper);
     }
 
+    public static void monitorTelemetry(MeterRegistry registry) {
```
Comment (christopheranderson) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/BridgeInternal.java]: ⚠ This is public API change exposing an unapproved public dependency, but we've documented that this public API shouldn't be depended on, so I'm not seeking approval at this time. Approved.

**Pull request merged**